### PR TITLE
Container Build Plan specification

### PR DESCRIPTION
**Pull request by chanseokoh**
\> [Preview](https://github.com/GoogleContainerTools/jib/blob/8ef257e49fad8509b9f0d816b48b3179c4a9f0d9/proposals/container-build-plan-spec.md) <

### CODE REVIEW

File: proposals/container-build-plan-spec.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+  "baseImage": "gcr.io/distroless/java@sha256:b715126ebd36e5d5c2fd730f46a5b3c3b760e82dc18dffff7f5498d0151137c9",
+  "architecture": "amd64",
+  "os": "linux",
+  "format": "Docker",
+  "created": "2011-12-03T22:42:05Z",
+
+  "config": {
+    "env": {
+      "KEY": "value",
+      "PATH": "/usr/sbin:/usr/bin:/sbin:/bin",
```
Comment (loosebazooka) [inline proposals/container-build-plan-spec.md]: should we allow passthrough from the baseImage env?
Could path be "PATH:$PATH" ?

File: proposals/container-build-plan-spec.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+      ]
+    },
+    {
+      "archive": {
+        "mediaType": "...",
+        "path": "/home/jane/misc/cacerts.tar",
+      }
+    },
+    {
+      "files": [
```
Comment (loosebazooka) [inline proposals/container-build-plan-spec.md]: `files` is not documented anywhere?

File: proposals/container-build-plan-spec.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    "entrypoint": ["/bin/bash", "-c"],
+    "cmd": [
+      "-x",
+      "set -o errexit ; echo \"\$0=$0 \$1=$1\" ; echo \"\$PATH=${PATH}\" ; exit 0",
+      "my-shell-name",
+      "first shell arg"
+    ],
+  },
+
+  "layers": [
```
Comment (loosebazooka) [inline proposals/container-build-plan-spec.md]: should each `"layers"` entry have a type and name?
could it be 
```
"layers": [
  {
    "type": "entries/v1"
    "name": "application classes"
    "values": [
      {
        ...
      },
      {
        ...
      },
     ]
  },
  {
    "type": "archive/v1"
    "name": "ca-certs"
    "values": {
      "mediaType": "...."
      "path": "/hasdfasdfafs/cacerts.tar"
    }
  }
}
```

File: proposals/container-build-plan-spec.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+    {
+      "entries": [
+        {
+          "src": "/home/jane/workspace/bin/Main.class",
```
Comment (loosebazooka) [inline proposals/container-build-plan-spec.md]: should we allow blob wildcards in here?

File: proposals/container-build-plan-spec.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+      "set -o errexit ; echo \"\$0=$0 \$1=$1\" ; echo \"\$PATH=${PATH}\" ; exit 0",
+      "my-shell-name",
+      "first shell arg"
+    ],
+  },
+
+  "layers": [
+    {
+      "entries": [
+        {
```
Comment (loosebazooka) [inline proposals/container-build-plan-spec.md]: could we have nested file-permission/owner configs? like in go/cbsp?

File: proposals/container-build-plan-spec.md (hunk 4 above)
Comment (loosebazooka) [reply proposals/container-build-plan-spec.md]: glob*

File: proposals/container-build-plan-spec.md (hunk 5 above)
Comment (chanseokoh) [reply proposals/container-build-plan-spec.md]: As discussed, we will consider such extension in the future.

File: proposals/container-build-plan-spec.md (hunk 4 above)
Comment (chanseokoh) [reply proposals/container-build-plan-spec.md]: As discussed, we will consider such extension in the future.

File: proposals/container-build-plan-spec.md (hunk 1 above)
Comment (chanseokoh) [reply proposals/container-build-plan-spec.md]: As discussed, we will consider such extension in the future.

File: proposals/container-build-plan-spec.md (hunk 4 above)
Comment (loosebazooka) [reply proposals/container-build-plan-spec.md]: Just to finish this comment chain, I misunderstood the purpose of this design. Glob support should come as a layer above this.

File: proposals/container-build-plan-spec.md (hunk 5 above)
Comment (loosebazooka) [reply proposals/container-build-plan-spec.md]: Again, at a different layer.

**Comment by chanseokoh** [comment]
Alright, let's merge. We can iterate over this as needed.

**Pull request merged**